raft/log.go

```diff
@@ -340,7 +340,7 @@ func (l *raftLog) mustCheckOutOfBounds(lo, hi uint64) error {
 	}
 
 	length := l.lastIndex() - fi + 1
-	if lo < fi || hi > fi+length {
```

Comment:
i feel the old checking is more readable with the panic message. no strong opinion. @heyitsanthony opinion?